### PR TITLE
Windows CI

### LINKED ISSUE #4: Setup CI for Windows and Linux 32

### PR DESCRIPTION
**Pull request by chancancode**
Closes #4, #17

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      RUST_HOST: x86_64-pc-windows-gnu
+    - RUBY_VERSION: 25
+      RUST_HOST: i686-pc-windows-gnu
+    - RUBY_VERSION: 25-x64
+      RUST_HOST: x86_64-pc-windows-gnu
+    - RUBY_VERSION: head-x86
+      RUST_HOST: i686-pc-windows-gnu
+    - RUBY_VERSION: head-x64
+      RUST_HOST: x86_64-pc-windows-gnu
+  # TODO: Maybe try `gnu` as well as the default `msvc`
```
Comment (wagenet) [inline .appveyor.yml]: This comment doesn’t make sense anymore.

File: .travis.yml [hunk 2]
```diff
@@ -26,3 +26,4 @@ before_install:
   - if [ ! -e "$HOME/.cargo/bin" ]; then curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain $RUST_VERSION -y; fi
   - export PATH="$HOME/.cargo/bin:$PATH"
   - rustup default $RUST_VERSION
```
Comment (wagenet) [inline .travis.yml]: Isn’t this redundant since we set a default in the install?

File: Rakefile [hunk 3]
```diff
@@ -61,8 +61,21 @@ namespace :build do
...
+      libcruby_sys_path = File.expand_path('lib')
+      libcruby_sys_name = 'cruby_sys'
+
+      cp File.expand_path("lib#{libcruby_sys_name}.so", libcruby_sys_path), File.expand_path("lib#{libcruby_sys_name}.dll", libcruby_sys_path)
+
+      sh "cargo rustc -- --cfg test -L #{libruby_path.inspect} -l #{libruby_name} -L #{libcruby_sys_path.inspect} -l #{libcruby_sys_name}"
+      cp "target/debug/libcruby_sys.dll", "target/debug/tests.#{Platform::DLEXT}"
+    else
+      sh 'cargo rustc -- --cfg test -C link-args="-Wl,-undefined,dynamic_lookup"'
+      cp "target/debug/liblibcruby_sys.#{Platform::LIBEXT}", "target/debug/tests.#{Platform::DLEXT}"
```
Comment (wagenet) [inline Rakefile]: Funny that we get the double lib on Linux but not Windows.

File: .travis.yml (hunk 2 above)
Comment (chancancode) [reply .travis.yml]: I'm not sure 100% how, but on appveyor, one of the builds where there is a .rustup cache, `rustup update` end up installing the msvc target, which is why I had to add the target to the default. I think we may not be caching wherever this default is being written to.

**Pull request merged**